app/gui/src/dashboard/layouts/Drive/Categories/Category.ts

```diff
@@ -202,64 +207,144 @@ export function areCategoriesEqual(a: Category, b: Category) {
 
 /** Whether an asset can be transferred between categories. */
 export function canTransferBetweenCategories(from: Category, to: Category) {
+  const operation = dropOperationBetweenCategories(from, to)
+
+  return operation !== 'cancel'
+}
+
+/**
+ * The drop operation to use when transferring assets between categories.
+ * @param from - The category to transfer from.
+ * @param to - The category to transfer to.
+ * @returns The drop operation to use.
+ */
+export function dropOperationBetweenCategories(from: Category, to: Category): DropOperation {
+  if (from.type === to.type) {
+    return 'cancel'
+  }
+
+  if (to.type === 'recent') {
+    return 'cancel'
+  }
+
+  if (isCloudCategory(from) || isCloudCategory(to)) {
+    if (isLocalCategory(from) || isLocalCategory(to)) {
+      return 'cancel'
+    }
+  }
+
   switch (from.type) {
     case 'cloud':
     case 'recent':
-    case 'team':
     case 'user': {
-      return to.type === 'trash' || to.type === 'cloud' || to.type === 'team' || to.type === 'user'
+      return 'move'
+    }
+    case 'team': {
+      if (to.type === 'trash') {
+        return 'move'
+      }
+
+      return 'copy'
     }
     case 'trash': {
-      // In the future we want to be able to drag to certain categories to restore directly
-      // to specific home directories.
-      return false
+      return 'move'
```

Comment:
even for `'trash' -> 'recent'`?

Reply:
moving to recent is not possible at all. that's we have this line above:
```
  if (to.type === 'recent') {
    return 'cancel'
  }
```